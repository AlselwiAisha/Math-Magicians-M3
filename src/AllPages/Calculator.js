import Calculator from '../Components/Calculator';

const Calc = () => (
  <Calculator />
);

export default Calc;
