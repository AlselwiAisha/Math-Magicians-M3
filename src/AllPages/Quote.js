import Quote from '../Components/Quotes';

const Quotes = () => (
  <Quote />
);

export default Quotes;
