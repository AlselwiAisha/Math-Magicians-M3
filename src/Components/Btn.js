import React from 'react';
import PropTypes from 'prop-types';

const getClassName = (valu) => {
  const classNames = {
    AC: 'btn1',
    '+/-': 'btn1',
    '%': 'btn1',
    'รท': 'btn2',
    x: 'btn2',
    '-': 'btn2',
    '+': 'btn2',
    '=': 'btn2',
    0: 'zero',
    '.': 'btn1',
  }; return classNames[valu] || 'num';
};
const Btn = ({ value }) => {
  const val = `${value}`;

  return (
    (
      <button type="button" className={`${getClassName(val)} btn`}>
        {val}
      </button>
    )
  );
};

Btn.propTypes = {
  value: PropTypes.string.isRequired,
};
export default Btn;
