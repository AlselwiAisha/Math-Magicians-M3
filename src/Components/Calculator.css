.calcmain {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
}

.calcmain h2 {
  font-size: 2.5rem;
  font-weight: 600;
  padding-bottom: 3rem;
}

.calcborder {
  margin: 2rem;
}

.clcBckgrnd {
  background-color: #f4f3f3;
  border-radius: 0.3rem;
  margin-top: 4rem;
  padding: 1rem;
  width: 30rem;
  box-shadow: 0 9px 15px -3px rgba(0, 0, 0, 0.1);
}

.clcScrn {
  font-size: 2rem;
  font-weight: 600;
  text-align: right;
  background-color: #858694;
  color: #e9f0f4;
  height: 4rem;
  margin-bottom: 0.1rem;
  padding: 0.8rem;
}

.btnsPad {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  gap: 0.1rem;
}

.btn {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #e0e0e0;
  color: #242424;
  font-size: 1.1rem;
  font-weight: 500;
  border: none;
  height: 3.5rem;
  padding: 10px;
  cursor: pointer;
}

.btn:hover {
  border: 2px solid #65737e;
}

.zero {
  grid-column: 1/3;
}

.btn2 {
  background: #f5913e;
}
