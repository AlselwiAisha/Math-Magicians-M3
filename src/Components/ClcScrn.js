import React from 'react';
/* eslint-disable react/prop-types */

const ClcScrn = ({ total, next, operation }) => {
  const tot = `${total}`;
  const nxt = `${next}`;
  const oprtion = `${operation}`;
  return (
    <>
      <div className="clcScrn">
        {tot}
        {oprtion}
        {nxt}
      </div>
    </>
  );
};

export default ClcScrn;
