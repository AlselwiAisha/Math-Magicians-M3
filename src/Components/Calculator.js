import React from 'react';
import './Calculator.css';
import ClcScrn from './ClcScrn';
import Btn from './Btn';

const Calculator = () => {
  const btnValues = [
    ['AC', '+/-', '%', 'รท'],
    ['7', '8', '9', 'x'],
    ['4', '5', '6', '-'],
    ['1', '2', '3', '+'],
    ['0', '.', '='],
  ];
  return (
    <div className="clcBckgrnd">
      <ClcScrn value="0" />
      <div className="btnsPad">
        {btnValues.flat().map((row) => (
          <Btn
            key={row.id}
            value={row}
          />
        ))}
      </div>
    </div>
  );
};
export default Calculator;
