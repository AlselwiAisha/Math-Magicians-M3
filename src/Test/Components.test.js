import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Btn from '../Components/Btn';
import renderWithRouter from '../Components/testPage';
import Layout from '../Components/Layout';
import NavBar from '../Components/Header';
import Calculator from '../Components/Calculator';
import Quote from '../Components/Quotes';
import ClcScrn from '../Components/ClcScrn';

describe('Btn component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Btn />);
    expect(container).toMatchSnapshot();
  });
});

describe('Calculator component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});

describe('ClcScrn component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<ClcScrn />);
    expect(container).toMatchSnapshot();
  });
});

describe('Layout component', () => {
  it('should match the snapshot', () => {
    const { container } = renderWithRouter(<Layout />);
    expect(container).toMatchSnapshot();
  });
});

describe('NavBar component', () => {
  it('should match the snapshot', () => {
    const { container } = renderWithRouter(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});
describe('Quote component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Quote />);
    expect(container).toMatchSnapshot();
  });
});
