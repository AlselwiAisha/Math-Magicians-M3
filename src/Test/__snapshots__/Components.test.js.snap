// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn component should match the snapshot 1`] = `
<div>
  <button
    class="num btn"
    type="button"
  >
    undefined
  </button>
</div>
`;

exports[`Calculator component should match the snapshot 1`] = `
<div>
  <div
    class="calcmain"
  >
    <h2>
      Let's do some math!
    </h2>
    <div
      class="calcborder"
    >
      <div
        class="clcBckgrnd"
      >
        <div
          class="clcScrn"
        >
          0
        </div>
        <div
          class="btnsPad"
        >
          <button
            class="num btn"
            type="button"
          >
            AC
          </button>
          <button
            class="num btn"
            type="button"
          >
            +/-
          </button>
          <button
            class="num btn"
            type="button"
          >
            %
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            รท
          </button>
          <button
            class="num btn"
            type="button"
          >
            7
          </button>
          <button
            class="num btn"
            type="button"
          >
            8
          </button>
          <button
            class="num btn"
            type="button"
          >
            9
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            x
          </button>
          <button
            class="num btn"
            type="button"
          >
            4
          </button>
          <button
            class="num btn"
            type="button"
          >
            5
          </button>
          <button
            class="num btn"
            type="button"
          >
            6
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            -
          </button>
          <button
            class="num btn"
            type="button"
          >
            1
          </button>
          <button
            class="num btn"
            type="button"
          >
            2
          </button>
          <button
            class="num btn"
            type="button"
          >
            3
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            +
          </button>
          <button
            class="zero btn"
            type="button"
          >
            0
          </button>
          <button
            class="num btn"
            type="button"
          >
            .
          </button>
          <button
            class="btn2 btn"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ClcScrn component should match the snapshot 1`] = `
<div>
  <div
    class="clcScrn"
  >
    undefined
    undefined
    undefined
  </div>
</div>
`;

exports[`Layout component should match the snapshot 1`] = `
<div>
  <div
    class="layout"
  >
    <nav
      class="nav"
    >
      <h2>
        Math Magicians
      </h2>
      <ul>
        <li
          class="link"
        >
          <a
            aria-current="page"
            class="navLink active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="link"
        >
          <a
            class="navLink"
            href="/Calculator"
          >
            Calculator
          </a>
        </li>
        <li
          class="link"
        >
          <a
            class="navLink"
            href="/Quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`NavBar component should match the snapshot 1`] = `
<div>
  <nav
    class="nav"
  >
    <h2>
      Math Magicians
    </h2>
    <ul>
      <li
        class="link"
      >
        <a
          aria-current="page"
          class="navLink active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="navLink"
          href="/Calculator"
        >
          Calculator
        </a>
      </li>
      <li
        class="link"
      >
        <a
          class="navLink"
          href="/Quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Quote component should match the snapshot 1`] = `
<div>
  <div
    class="Container"
  >
    <p
      class="loading"
    >
      Loading ...
    </p>
  </div>
</div>
`;
